test/unit/account_info/test_account_info.py

```diff
@@ -61,3 +71,268 @@ def test_s3_api_url(self, s3_api_url):
+        if self.PERSISTENCE:
+            assert 'bucket-0' == self._make_info(
+            ).get_bucket_id_or_none_from_bucket_name('my-bucket')
+            assert 'my-bucket' == self._make_info(
+            ).get_bucket_name_or_none_from_bucket_id('bucket-0')
+        account_info.remove_bucket_name('my-bucket')
+        assert account_info.get_bucket_id_or_none_from_bucket_name('my-bucket') is None
+        assert account_info.get_bucket_name_or_none_from_bucket_id('bucket-0') is None
+        if self.PERSISTENCE:
+            assert self._make_info().get_bucket_id_or_none_from_bucket_name('my-bucket') is None
+            assert self._make_info().get_bucket_name_or_none_from_bucket_id('bucket-0') is None
+
+    def test_refresh_bucket(self):
+        account_info = self._make_info()
+        assert account_info.get_bucket_id_or_none_from_bucket_name('my-bucket') is None
+        assert account_info.get_bucket_name_or_none_from_bucket_id('a') is None
+        bucket_names = {'a': 'bucket-0', 'b': 'bucket-1'}
+        account_info.refresh_entire_bucket_name_cache(bucket_names.items())
+        assert 'bucket-0' == account_info.get_bucket_id_or_none_from_bucket_name('a')
+        assert 'a' == account_info.get_bucket_name_or_none_from_bucket_id('bucket-0')
+        if self.PERSISTENCE:
+            assert 'bucket-0' == self._make_info().get_bucket_id_or_none_from_bucket_name('a')
+            assert 'a' == self._make_info().get_bucket_name_or_none_from_bucket_id('bucket-0')
+
+    @pytest.mark.apiver(to_ver=1)
+    def _test_account_info_up_to_v1(self):
+        account_info = self._make_info()
+        account_info.set_auth_data(
+            'account_id',
+            'account_auth',
+            'https://api.backblazeb2.com',
+            'download_url',
+            100,
+            'app_key',
+            'realm',
+            application_key_id='key_id'
+        )
+
+        object_instances = [account_info]
+        if self.PERSISTENCE:
+            object_instances.append(self._make_info())
+        for info2 in object_instances:
+            assert 'account_id' == info2.get_account_id()
+            assert 'account_auth' == info2.get_account_auth_token()
+            assert 'https://api.backblazeb2.com' == info2.get_api_url()
+            assert 'app_key' == info2.get_application_key()
+            assert 'key_id' == info2.get_application_key_id()
+            assert 'realm' == info2.get_realm()
+            assert 100 == info2.get_minimum_part_size()
+            assert info2.is_same_key('key_id', 'realm')
+            assert not info2.is_same_key('key_id', 'another_realm')
+            assert not info2.is_same_key('another_key_id', 'realm')
+            assert not info2.is_same_key('another_key_id', 'another_realm')
+
+    @pytest.mark.apiver(from_ver=2)
+    def _test_account_info_v2(self):
+        account_info = self._make_info()
+        account_info.set_auth_data(
+            account_id='account_id',
```

Comment:
if the order of arguments (especially optional) is accidentally messed up after v2 release (even in v2+N in the future), then this test is going to pass, because it escapes the verification of the order of arguments. On the other hand, if we switch it to positional arguments and one argument is renamed accidentally, then the test will pass but invoking the function with its keyword arguments is going to fail.

I think we should call it twice, once with positional arguments and once with named, to be sure we did not mess up the external interface... but that goes for everything, not just for this method. Not even for b2sdk external interface, but for backward compatibility of every module (not just a python module, java etc has the same problem). It might be a good idea to think about this to harden the interface somehow. Lets discuss it separately though (not in this PR).